pocs/utils/messaging.py

```diff
@@ -14,14 +15,79 @@
 
 
 class PanMessaging(object):
-
-    """Messaging class for PANOPTES project. Creates a new ZMQ
-    context that can be shared across parent application.
-
+    """Provides messaging services within a PANOPTES robotic telescope.
+
+    Supports broadcasting messages from publishers (e.g. a POCS or
+    ArduinoIO class instance) to subscribers (also typically class
+    instances). The publishers and subscribers may be in the same
+    process, or in separate processes. The messages all go through
+    a message forwarder; this is a process which listens for messages
+    from all publishers on one TCP port and forwards each message to
+    all subscribers that are connected to a second TCP port.
+
+    Do not create PanMessaging instances directly. Publishers should
+    call PanMessaging.create_publisher to create an instance of
+    PanMessaging, on which they can then call send_message.
+    Subscribers should call PanMessaging.create_subscriber to create
+    an instance of PanMessaging, on which they can then call
+    receive_message.
+
+    Messages are sent to channels, a name that can be used to allow
+    a high-level partitioning of messages. A channel name may not
+    include whitespace. Among the currently used channel names are:
+
+      * PANCHAT (sent from POCS.say)
+      * PAWS-CMD (sent from PAWS websockets.py)
+      * POCS (sent by class POCS)
+      * POCS-CMD (sent by class POCS)
+      * STATUS (sent by class POCS)
+      * weather (from peas/sensors.py)
+      * environment (from peas/sensors.py)
+      * telemetry:commands (in ArduinoIO... new)
+      * camera:commands (in ArduinoIO... new)
+
+    And some other channels are used in tests:
+
+      * TEST-CHANNEL (test_messaging.py)
+      * RUNNING (test_pocs.py)
+      * POCS-CMD (test_pocs.py)
+
+    The method receive_message will return messages from all channels;
```

Comment:
While this is true, the intended usage is to pass the channel to the `create_subcriber` and then no filtering is needed by the user. This should be the recommended behaviour.